sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -90,6 +90,11 @@ private[sql] case class JDBCRelation(
 
   override val schema: StructType = JDBCRDD.resolveTable(url, table, properties)
 
+  // Check if JDBCRDD.compileFilter can accept input filters
+  override def unhandledFilters(filters: Array[Filter]): Array[Filter] = {
+    filters.filterNot(JDBCRDD.compileFilter(_) != null)
```

Comment:
Nit: It's a little bit weird to use double-negation here when `filters.filter(JDBCRDD.compileFilter(_) == null)` is perfectly OK.